maa-cli/src/dirs.rs

```diff
@@ -394,6 +400,166 @@ fn ensure_name(name: &str) -> &str {
+                let mut logs = vec![log().join("asst.log"), log().join("asst.bak.log")];
+                logs.extend(log_files);
+                logs
+            }
+            CleanupTarget::Map => {
+                vec![
+                    debug.join("drops"),
+                    debug.join("map"),
+                    debug.join("other"),
+                    debug.join("Roguelike"),
+                ]
+            }
+        }
+    }
+}
+
+pub fn cleanup<T>(targets: &[T]) -> Result<()>
+where
+    T: PathProvider,
+{
+    let targets_to_use: Vec<&dyn PathProvider> = if targets.is_empty() {
+        vec![
+            &CleanupTarget::CliCache,
+            &CleanupTarget::Avatars,
+            &CleanupTarget::Log,
+            &CleanupTarget::Map,
+        ]
+    } else {
+        targets.iter().map(|x| x as &dyn PathProvider).collect()
+    };
+
+    let trash_list: Vec<PathBuf> = targets_to_use
+        .iter()
+        .flat_map(|target| target.get_path())
+        .collect();
+
+    trash_list.iter().enumerate().for_each(|(i, p)| {
+        println!("{}. {}", i + 1, p.display());
+    });
+
+    if !clear_inquiry("files or folders mentioned above?")? {
+        println!("No files or folders have been deleted.");
+        return Ok(());
+    }
+
+    let mut has_err = false;
+    for path in trash_list {
+        print!("Delete {}... ", path.display());
+        let exclude = if path == cache() {
+            // Keep the latest packages
+            Some(run::core_version()?)
+        } else {
+            None
+        };
+
+        let result = del_item(path.as_path(), exclude);
+
+        match result {
+            Err(e) => {
+                println!("\x1B[31m{}\x1B[0m", e);
+                has_err = true;
+            }
+            Ok(_) => {
+                println!("\x1B[34mDone.\x1B[0m");
+            }
+        }
+    }
+
+    if !has_err {
+        Ok(())
+    } else {
+        Err(anyhow!("At least one path has not been deleted."))
+    }
+}
+
+fn del_item(path: &Path, exclude: Option<&str>) -> Result<()> {
+    let exclude_logic = |entry: DirEntry| match exclude {
+        Some(str) => {
+            let binding = entry.file_name();
+            if !binding.to_str()?.contains(str) {
+                Some(entry.path())
+            } else {
+                None
+            }
+        }
+        None => Some(entry.path()),
+    };
+
+    if path.is_file() {
+        std::fs::remove_file(path)?;
+        return Ok(());
+    }
+
+    let filtered_dir_list: Vec<PathBuf> = fs::read_dir(path)?
+        .filter_map(|e| e.ok())
+        .filter_map(exclude_logic)
+        .collect();
+    log::debug!("{:?}", filtered_dir_list);
+
+    if filtered_dir_list.is_empty() {
+        return Err(anyhow!("Folder is empty."));
+    }
+
+    for path in filtered_dir_list {
+        if path.is_file() {
+            std::fs::remove_file(path)?;
+        } else {
+            std::fs::remove_dir_all(path)?
+        }
+    }
```

Comment:
I prefer don't use debug format.

```suggestion
    if filtered_dir_list.is_empty() {
        return Err(anyhow!("Folder is empty."));
    }

    for path in filtered_dir_list {
        log::debug!("Deleting {}", path.display());
        if path.is_file() {
            std::fs::remove_file(path)?;
        } else {
            std::fs::remove_dir_all(path)?
        }
    }
```